subgraphs/notional-finance/src/setters/transactions.ts

```diff
@@ -298,6 +298,13 @@ export function createDeposit(
 
   deposit.save();
 
+  // update lending rates for deposit trades
+  // used for supply side revenue calculations
+  const mkt = getOrCreateMarket(event, market.id);
+  const interestRate = getOrCreateInterestRate(market.rates[0]).rate;
+  mkt.lendRates!.push(interestRate);
```

Comment:
you need to add to the array like this:

```
const rates = mkt.lendRates;
rates.push(interestRate);
mkt.lendRates = rates;
```

Reply:
I am curious how big these arrays will be, this slows down syncing a lot. I know it is not really an issue for notional, but better to form good subgraph habits.

Another option is to store the sum of all rates and then the number and then do the average after.

Reply:
you need to add to the array like this
I forget why we need to do it this way. Why would a `push` directly on market entity parameter not work since it should update the array and return the length of the array?

Reply:
> I am curious how big these arrays will be, this slows down syncing a lot. I know it is not really an issue for notional, but better to form good subgraph habits.
I don’t think they would be bigger than 100 for most of the markets.

> Another option is to store the sum of all rates and then the number and then do the average after.
Good idea. Let’s go ahead with this. (I don’t think we need to be able to see all the lend rates so this should be fine).

Reply:
https://github.com/messari/subgraphs/blob/master/docs/ERRORS.md#array-sorting-when-querying

And I can't find the graph docs for this rn, but that is they talk about it too if you want to know more

Reply:
Got it, we wouldn't have been impacted by this because we are taking an average and the ordering of the items don't matter. 

We've also moved away to maintaining the sum instead of array.

Reply:
Ah I linked the wrong docs. The issue is not sorting, it is that you cannot `.push()` to an array on an entitiy. You have to save it in a temp var first.

Again, doesn't matter since we aren't using arrays anymore. Feel free to resolve